src/coreclr/minipal/Unix/doublemapping.cpp

```diff
@@ -53,15 +53,22 @@ bool VMToOSInterface::CreateDoubleMemoryMapper(void** pHandle, size_t *pMaxExecu
 
 #ifdef TARGET_FREEBSD
     int fd = shm_open(SHM_ANON, O_RDWR | O_CREAT, S_IRWXU);
-#elif defined(TARGET_SUNOS) // has POSIX implementation
-    char name[24];
-    sprintf(name, "/shm-dotnet-%d", getpid());
-    name[sizeof(name) - 1] = '\0';
-    shm_unlink(name);
-    int fd = shm_open(name, O_RDWR | O_CREAT | O_EXCL | O_NOFOLLOW, 0600);
-#else // TARGET_FREEBSD
+#elif defined(TARGET_LINUX)
     int fd = memfd_create("doublemapper", MFD_CLOEXEC);
-#endif // TARGET_FREEBSD
+#else
+    int fd = -1;
+#endif
+
+    // POSIX fallback
+    if (fd == -1)
+    {
+        char name[24];
+        sprintf(name, "/shm-dotnet-%d", getpid());
+        name[sizeof(name) - 1] = '\0';
+        shm_unlink(name);
+        fd = shm_open(name, O_RDWR | O_CREAT | O_EXCL | O_NOFOLLOW, 0600);
+        shm_unlink(name);
+    }
```

Comment:
Shouldn't this block be in `#else`?

Reply:
No, it should not, as stated in https://github.com/dotnet/runtime/pull/93907#discussion_r1371114236.

Even for OSes with platform-specific ways to get a special memory file open, should their method fail, we can still always use the POSIX fallback.